api/email/emailModel.js

```diff
@@ -113,4 +114,27 @@ async function saveResetCode(uid, code)
     {
         console.log(ex); return -1;
     }
+}
+
+// Regex checking if the username only contains letters and numbers
+async function userNamePattern(username) 
+{
+    {
+        const codenamePattern = /^[A-Za-z0-9]*$/;
+        try 
```

Comment:
Shouldn't need try/catch, test shouldn't have any uncaught throws - we should probably add some type validation to the forms though thinking about it now!